### PR TITLE
feat: wishlist

### PR DESCRIPTION
**Pull request by shristabaruah**
- Add and remove from wishlist
- cart functionality is not added

### CODE REVIEW

File: src/components/product-card/product-card.jsx [hunk 1]
```diff
@@ -1,45 +1,78 @@
...
+    discount,
+  } = props;
 
-    );
-}
-export { ProductCard };
\ No newline at end of file
+  const { wishlist, addToWishlist, removeWishlist } = useWishlist();
+  //   const itemExistsInWishlist = wishlist.some((item)=>item._id === _id);
+  //   console.log(itemExistsInWishlist)
```
Comment (RahullRawat) [inline src/components/product-card/product-card.jsx]: You can remove this unwanted comments

File: src/contexts/wishlist-context.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  useEffect(
+    () =>
+      (async () => {
+        try {
+          const response = await axios.get("/api/user/wishlist", {
+            headers: { authorization: token },
+          });
+         setWishlist(response.data.wishlist)
+        } catch (e) {
+          console.log("error:", e.response);
```
Comment (RahullRawat) [inline src/contexts/wishlist-context.js:20]: Remove this console.log

File: src/contexts/wishlist-context.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    []
+  );
+
+  const addToWishlist = async ( products)=>{
+    try {
+        const response = await axios.post("/api/user/wishlist",{product:products},{headers:{authorization:token}});
+        setWishlist(response.data.wishlist)
+       
+        }catch(e){
+            console.log("error:",e)
```
Comment (RahullRawat) [inline src/contexts/wishlist-context.js:32]: Remove this also

File: src/contexts/wishlist-context.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        }
+      }
+      const removeWishlist = async (id)=>{
+        try{
+            const response =await axios.delete(`/api/user/wishlist/${id}`,{headers:{authorization:token}})
+            setWishlist(response.data.wishlist)
+           
+        }catch(e){
+            console.log("error:",e)
```
Comment (RahullRawat) [inline src/contexts/wishlist-context.js:41]: Remove this also

**Review by RahullRawat — APPROVED** [review]
Wishlist functionality is working fine . Great work @shristabaruah

File: src/pages/Homepage/home.jsx [hunk 5]
```diff
@@ -71,7 +72,8 @@ const Home = () => {
         <div className="product-card">
           {bestseller.map((p) => (
```
Comment (Kevin-Solomon) [inline src/pages/Homepage/home.jsx:73]: You can try destructuring here instead of using ```p```

**Review by Kevin-Solomon — APPROVED** [review]

File: src/components/product-card/product-card.jsx (hunk 1 above)
Comment (shristabaruah) [reply src/components/product-card/product-card.jsx]: yes will do that

**Pull request merged**